### PR TITLE
Speed up "Install package" task

### LINKED ISSUE #155: [2pt] Too long "Install package" task
I have 100 Tarantool instances in a cluster. When I deploy them via ansible-cartridge, it takes a very long time to complete the "Install package" task, since it is performed for each instance. Moreover, this task is performed even despite the fact that it is written in the logs that task is skipped for all instances.

Let's define the host list once instead of a hundred times.

### PR DESCRIPTION
**Pull request by hackallcode**
Closes #155

### Changes

Before this patch, instances for the "Install Package" task were selected on each instance, it was too long.

Now the list of instances for installing a package is selected once for all.
This made it possible to speed up the task at times for a large cluster.

### Complexity and Time

Before this patch, the complexity of calculating the list of instances was `O(N^2)`, now it is `O(N)`.
For 100 instances, this gives a 10x time reduction, from 60 seconds to 5 seconds.

### CODE REVIEW

File: tasks/main.yml [hunk 1]
```diff
@@ -42,12 +42,18 @@
     - cartridge-replicasets
     - cartridge-config
 
+- name: Calculate hosts list
```
Comment (dokshina) [inline tasks/main.yml]: Please, don't name machines `hosts`. Ansible `hosts` are instances in this role, I think we should use "machine" word to underline that its physical servers, not instances.

File: filter_plugins/filter_hosts.py [hunk 2]
```diff
@@ -1,5 +1,8 @@
-def get_machine_identifier(v):
-    return v['ansible_host'] if 'ansible_host' in v else v['inventory_hostname']
+def get_machine_hostname(v):
+    if 'ansible_host' in v:
+        return v['ansible_host']
+
+    raise Exception(f'Instance {v["inventory_hostname"]} has not "ansible_host" option!')
```
Comment (dokshina) [inline filter_plugins/filter_hosts.py:5]: `has not` -> `hasn't`? `doesn't have`? `has not`?

I'm not sure it's wrong.

Comment (hackallcode) [reply filter_plugins/filter_hosts.py:5]: I think everything is right

File: tasks/main.yml (hunk 1 above)
Comment (hackallcode) [reply tasks/main.yml]: Please, check new variant of variables names

File: unit/test_tags.py [hunk 3]
```diff
@@ -45,6 +45,7 @@ def test_without_tags(self):
             'Set remote_user for delegated tasks',
             'Validate config',
             'Set instance facts',
+            'Calculate machine hosts list',
```
Comment (dokshina) [inline unit/test_tags.py]: Get physical machines list?
Get one instance from each physical machine to install package?

**Review by dokshina — COMMENTED** [review]
Do you have some measurements that shows advantages of a new approach?

File: unit/test_tags.py (hunk 3 above)
Comment (hackallcode) [reply unit/test_tags.py]: Get one instance from each physical machine

File: tasks/main.yml [hunk 4]
```diff
@@ -42,12 +42,18 @@
     - cartridge-replicasets
     - cartridge-config
 
+- name: Get one instance from each physical machine
+  set_fact:
+    machine_hosts_list: '{{ hostvars | get_one_not_expelled_instance_for_machine(play_hosts) }}'
```
Comment (dokshina) [inline tasks/main.yml]: Maybe `machine_hostnames`?

**Comment by dokshina** [comment]
I think this fix can be mentioned in Changelog.

File: tasks/main.yml (hunk 4 above)
Comment (hackallcode) [reply tasks/main.yml]: But it's not a hostnames (e.g. `tarantool.io`), it's names of Ansible hosts (e.g. `my_core_instance`)

**Comment by hackallcode** [comment]
> Do you have some measurements that shows advantages of a new approach?

Before this patch, the complexity of calculating the list of instances was `O(N^2)`, now it is `O(N)`. For 100 instances, this gives a 10x time reduction

File: tasks/main.yml (hunk 4 above)
Comment (dokshina) [reply tasks/main.yml]: So, it's smth like `instances_to_install_package`?

Comment (hackallcode) [reply tasks/main.yml]: `single_instances_for_each_machine`

**Review by dokshina — APPROVED** [review]

**Pull request merged**